src/amber/router/params.cr

```diff
@@ -31,11 +32,37 @@ module Amber::Router
       end
     end
 
-    def upgrade_request_method!
-      if params[METHOD]?
-        method = params[METHOD]
-        request.method = method.upcase if OVERRIDE_METHODS.includes?(method)
-      end
+    # Adds Request Method Override support to the framework.
+    # Param supported
+    # - *_method* can be passed as a form or url param
+    #
+    # HTTP Headers supported:
+    # - *X-HTTP-Method-Override* (Google/GData)
+    #
+    # The convention has been established that the GET and HEAD methods SHOULD NOT
+    # have the significance of taking an action other than retrieval.
+    #
+    # These methods ought to be considered "safe". This allows user agents to
+    # represent other methods, such as POST, PUT and DELETE, in a special way,
+    # so that the user is made aware of the fact that a possibly unsafe action
+    # is being requested.
+    #
+    # Read RFC 2616 - HTTP 1.1 section 9.1.1
+    # (https://tools.ietf.org/html/rfc2616#section-9.1)
+    #
+    # In other words, if you are tempted to use a GET to simulate a PUT or DELETE,
+    # don't do it. Use a POST instead.
+    def override_request_method!
+      # If the current request method is not GET or POST it means that it was
+      # already overrided
+      return unless %(GET POST).includes? request.method
```

Comment:
The code on line 58 is not restricting but bypassing if the current request is something other than a GET or POST, meaning that the request was upgraded before reaching to the amber server either by a proxy or http server that supports upgrades. You can still make GET request with `_method` and/or `X-HTTP-Method-Override` 

What this line is trying to avoid is to have a request.method = `PUT` with and override to some other method.

See line for GET support https://github.com/Amber-Crystal/amber/pull/117/files#diff-707c9269bb28ed65da046758b2df5c48R18 and https://github.com/Amber-Crystal/amber/pull/117/files#diff-707c9269bb28ed65da046758b2df5c48R56